src/change/sdp.rs

```diff
@@ -479,13 +512,38 @@ fn create_offer(rtc: &mut Rtc, changes: &Changes) -> SdpOffer {
     sdp.into()
 }
 
-fn add_ice_details(rtc: &mut Rtc, sdp: &Sdp) -> Result<(), RtcError> {
-    if let Some(creds) = sdp.ice_creds() {
-        rtc.ice.set_remote_credentials(creds);
-    } else {
+fn add_ice_details(
+    rtc: &mut Rtc,
+    sdp: &Sdp,
+    pending: Option<&SdpPendingOffer>,
+) -> Result<(), RtcError> {
+    let Some(creds) = sdp.ice_creds() else {
         return Err(RtcError::RemoteSdp("missing a=ice-ufrag/pwd".into()));
+    };
+
+    // If we are handling an **offer** from the remote, differing ICE credentials indicate an ICE
+    // restart initiated by the remote.
+    //
+    // If we are handling an **answer** from the remote, differing ICE credentials indicate an
+    // acceptance of an ICE restart we requested.
+    let ice_restart = match rtc.ice.remote_credentials() {
+        Some(v) => *v != creds,
+        None => false,
+    };
+
+    // If we initiated the ICE Restart, the credentials are in the
+    // pending SDP changes. Otherwise we need to create new ones now.
+    let (new_local_creds, keep_local_candidates) = pending
+        .and_then(|p| p.ice_restart())
+        .unwrap_or_else(|| (IceCreds::new(), true));
```

Comment:
This only needs to happen if we are doing `ice_restart`, move to inside the if-statement?

Reply:
Good catch, that wasn't intentional